File: app/src/main/java/closer/vlllage/com/closer/api/models/PhoneResult.java
package closer.vlllage.com.closer.api.models;

import java.util.List;

import closer.vlllage.com.closer.store.models.Phone;

public class PhoneResult extends ModelResult {
    public List<Double> geo;
    public String name;
    public String photo;
    public String status;
    public Boolean active;

    public static Phone from(PhoneResult phoneResult) {
        return updateFrom(new Phone(), phoneResult);
    }

    public static Phone updateFrom(Phone phone, PhoneResult phoneResult) {
        phone.setId(phoneResult.id);
        phone.setUpdated(phoneResult.updated);

        if (phoneResult.geo != null && phoneResult.geo.size() == 2) {
            phone.setLatitude(phoneResult.geo.get(0));
            phone.setLongitude(phoneResult.geo.get(1));
        }

        phone.setName(phoneResult.name);
        phone.setStatus(phoneResult.status);
        phone.setPhoto(phoneResult.photo);

        return phone;
    }
}

File: app/src/main/java/closer/vlllage/com/closer/api/models/PhoneResult.kt
package closer.vlllage.com.closer.api.models

class PhoneResult : ModelResult() {
    var geo: List<Double>? = null
    var geoIsApprox: Boolean? = null
    var name: String? = null
    var photo: String? = null
    var status: String? = null
    var introduction: String? = null
    var offtime: String? = null
    var occupation: String? = null
    var history: String? = null
    var active: Boolean? = null
    var verified: Boolean? = null
    var goals: List<GoalResult>? = null
    var lifestyles: List<LifestyleResult>? = null
}
